src/main/java/com/lovetropics/gamemodebuild/message/SetGamemodeBuildSlotPacket.java

```diff
@@ -0,0 +1,96 @@
+package com.lovetropics.gamemodebuild.message;
+
+import com.lovetropics.gamemodebuild.GBConfigs;
+import com.lovetropics.gamemodebuild.GamemodeBuild;
+import com.lovetropics.gamemodebuild.container.BuildContainer;
+import com.lovetropics.gamemodebuild.state.GBPlayerStore;
+import com.lovetropics.gamemodebuild.state.GBServerState;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.RegistryAccess;
+import net.minecraft.core.component.DataComponents;
+import net.minecraft.network.RegistryFriendlyByteBuf;
+import net.minecraft.network.codec.ByteBufCodecs;
+import net.minecraft.network.codec.StreamCodec;
+import net.minecraft.network.protocol.PacketUtils;
+import net.minecraft.network.protocol.common.custom.CustomPacketPayload;
+import net.minecraft.network.protocol.game.ServerboundSetCreativeModeSlotPacket;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.SimpleMenuProvider;
+import net.minecraft.world.flag.FeatureFlagSet;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.component.CustomData;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.neoforged.neoforge.network.handling.IPayloadContext;
+
+import java.util.List;
+
+public record SetGamemodeBuildSlotPacket(short slotNum, ItemStack itemStack)  implements CustomPacketPayload {
```

Comment:
Although this is what creative mode does, and we do have filtering here (thanks for adding that in :sweat_smile:), it would be really nice to avoid this kind of client -> server item request. Particularly, from experience of how flimsy this can be in creative mode already - and how dangerous sending an ItemStack client->server can be in general, for all the kinds of weirdly nested/recursive structures that can be constructed. 

In the latest snapshots, this is something Vanilla actually handles totally server-driven - I think it shouldn't be too tricky to do a similar kind of thing, where the client just sends a "hey, pick this block for me" packet and the server does all the logic and tells the client about what item it gets. (Block entity serialisation on the client also is a bit of a scary one :sweat_smile:)

Reply:
oh thats a good point and wouldnt be too tall a task to implement, Ill take a look later at switching it over to a blockpos rather than item stack target. Luckily the rest of the code can be kept the same :)